Create base image

nathanfranklin opened the pull request.

nathanfranklin committed:
Create base image for providing applications

nathanfranklin committed:
Add support for various screen depth (DISPLAY_SCREEN_DEPTH)

env varible to control this is DISPLAY_SCREEN_DEPTH. If not set, it defaults to 16.

nathanfranklin committed:
Update readme

nathanfranklin committed:
Fix spellings

nathanfranklin committed:
Set default resolution

nathanfranklin committed:
Add check for 'mydata' directory

nathanfranklin committed:
Fix readme

nathanfranklin committed:
Do not check for nginx.key as could have different name

Earlier versions, an nginx.key was always at /etc/nginx/ssl/

nathanfranklin committed:
Fix DS certs and add startup check

nathanfranklin committed:
Improve README

nathanfranklin committed:
Fix check for mydata dir

rstijerina commented on Dockerfile.amd64:
```diff
@@ -113,9 +110,19 @@ COPY rootfs /
+RUN groupadd --gid 816877 G-816877
+RUN useradd --uid 458981 --create-home --shell /bin/bash --user-group --groups G-816877,adm,sudo ubuntu && chown ubuntu:G-816877 /home/ubuntu
+RUN usermod -g G-816877 ubuntu
+RUN mkdir -p /home/ubuntu/.config/pcmanfm/LXDE/ && cp /usr/local/share/doro-lxde-wallpapers/desktop-items-0.conf /home/ubuntu/.config/pcmanfm/LXDE/ && chown -R ubuntu:G-816877 /home/ubuntu/.config
```
perhaps these group and user specific commands should be defined by the app that uses this base, since they may need different ids depending on the portal using the app

nathanfranklin replied:
Yes, made env variables for that. They currently default to designsafe user and gid.

rstijerina commented on rootfs/startup.sh:
```diff
@@ -43,21 +47,33 @@ cp -r /root/{.gtkrc-2.0,.asoundrc} ${HOME}
 sed -i -e "s|%USER%|$USER|" -e "s|%HOME%|$HOME|" /etc/supervisor/conf.d/supervisord.conf
 
+# check for 'mydata' working directory
+if [ ! -d "$HOME/mydata/" ]; then
```
maybe we can find a way to make the app run without any volume mounts

nathanfranklin replied:
added an environmental variable for that

rstijerina commented on rootfs/startup.sh:
```diff
@@ -43,21 +47,33 @@ cp -r /root/{.gtkrc-2.0,.asoundrc} ${HOME}
-    echo "* enable SSL"
-	sed -i 's|#_SSL_PORT_#\(.*\)443\(.*\)|\1'$SSL_PORT'\2|' /etc/nginx/sites-enabled/default
-	sed -i 's|#_SSL_PORT_#||' /etc/nginx/sites-enabled/default
+if [ -n "$SSL_PORT" ]; then
```
could we configure ssl in the app that uses this base? or are we going to have several base images for each ssl cert

nathanfranklin replied:
fixed

nathanfranklin committed:
Fix setting DISPLAY_SCREEN_DEPTH

This was updated from default 16 to 24 so this sed
command needed to be updated

nathanfranklin committed:
App working directory now defaults to /home/ubuntu

Can be configured with GUI_APPLICATION_DIRECTORY

nathanfranklin committed:
Make password more generic

nathanfranklin committed:
Make cert/keys more generic

nathanfranklin committed:
Make user uid and gid be configurable

nathanfranklin committed:
Run application directly and not in xterm

nathanfranklin committed:
Add note about application.sh permissions to readme

nathanfranklin committed:
Fix startup.sh

nathanfranklin commented:
Closing as replaced by #2
